AutoGraphTests/Stub.swift

```diff
@@ -112,3 +129,102 @@ extension String {
         return components.filter { !$0.isEmpty }.joined(separator: " ")
     }
 }
+
+extension InputStream {
+    func readfully() -> Data {
+        var result = Data()
+        var buffer = [UInt8](repeating: 0, count: 4096)
+        
+        self.open()
+        defer { self.close() }
+        
+        var amount = 0
+        repeat {
+            amount = read(&buffer, maxLength: buffer.count)
+            if amount > 0 {
+                result.append(buffer, count: amount)
+            }
+        } while amount > 0
+        
+        return result
+    }
+}
+
+class MockURLProtocol: URLProtocol {
```

Comment:
`MockURLProtocol` hijacks the URLProtocol that a request will use when a stub is registered. Everything ends up going through `startLoading`.